Issue 5700 - SRCH performance drop 2.0 vs 1.4

tbordaz opened the pull request:
Bug description:
	The commit 40563667e12da3042ea5a0e3771c1e1db1dafa06
	force the evaluation of the filter before returning
	entries. This evaluation can be skipped in case
	of indexed search and filter syntax without OR
	(sr_flags ~SR_FLAG_MUST_APPLY_FILTER_TEST).
	The consequence of the fix is a severe drop of
	search performance.
	The testcase provided with the commit are PASS
	without part of the original fix

Fix description:
	Revert parts of the fix.
	This patch intend to identify new testcase
	that confirm the need to force the evaluation
	of the filter

relates: #5700

Reviewed by:

Firstyear commented on ldap/servers/slapd/filterentry.c:
```diff
@@ -770,12 +770,6 @@ slapi_vattr_filter_test_ext(
 
-    if (only_check_access != 0) {
-        slapi_log_err(SLAPI_LOG_ERR, "slapi_vattr_filter_test_ext",
-            "⚠️  DANGER ⚠️  - only_check_access mode is BROKEN!!! YOU MUST CHECK ACCESS WITH FILTER MATCHING");
```
This comment was not incorrect. This is broken, and can lead to access control bypasses.

Firstyear commented on ldap/servers/slapd/back-ldbm/ldbm_search.c:
```diff
@@ -1753,15 +1753,15 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
                                       "Bypassing filter test for %s\n", slapi_entry_get_dn_const(e->ep_entry));
                         if (ACL_CHECK_FLAG) {
-                            filter_test = slapi_vattr_filter_test(pb, e->ep_entry, filter_intent, ACL_CHECK_FLAG);
+                            filter_test = slapi_vattr_filter_test_ext(pb, e->ep_entry, filter_intent, ACL_CHECK_FLAG, 1);
```
we can't set this. This allows access control bypasses and is a CVE.

Firstyear reviewed (changes requested):
I know you want to try to fix this to remove the extra filter tests, but I had to re-add them because they are correct. There were access control bypasses and query corruptions here.

I know it sucks that it affects performance, but setting this flags just undoes all that work I did to fix these to be correct and secure. 

The existing code is *already* the correct set of filter tests and access controls to prevent these bypasses. 

I can't approve this, because it would introduce vulnerabilities. 

If you want to actually improve this, then *remove grok_filter*.

Grok filter is *fundamentally broken* and always forces filter tests even if they are not needed!

tbordaz committed:
Issue 5700 - SRCH performance drop 2.0 vs 1.4

Bug description:
        The commit 40563667e12da3042ea5a0e3771c1e1db1dafa06
        force the evaluation of the filter before returning
        entries. This evaluation can be skipped in case
        of indexed search and filter syntax without OR
        (sr_flags ~SR_FLAG_MUST_APPLY_FILTER_TEST).
        The consequence of the fix is a severe drop of
        search performance.
        The testcase provided with the commit are PASS
        without part of the original fix

Fix description:
        Revert parts of the fix.
        This patch intend to identify new testcase
        that confirm the need to force the evaluation
        of the filter

relates: #5700

Reviewed by:
